### PR TITLE
IAM Policy Required Bindings Constraint v1

### PR DESCRIPTION
**Pull request by hshin-g**
### IAM Policy Required Bindings constraint for #267 

This PR creates a IAM Policy Required Bindings constraint that checks for the bindings of IAM policies and checks whether all of the user defined required members are included. If not, generate a violation.

A sample playground for this constraint is provided [here](https://play.openpolicyagent.org/p/707GzQieZ1).

### This constraint provides the following features:

1. **asset_type**
- Restrict which asset type this policy applies to.
  - E.g. cloudresourcemanager.googleapis.com/Project would restrict to only Project resources
2. **role**
- The role to restrict bindings for. Wildcards (*) supported.
  - E.g. roles/owner would only look at bindings for the owner role.
  - E.g. roles/* would look at all roles available.
3. **members**
- Members that are required for the given role. This can be specific users, groups, or domains. Wildcards (*) supported.
  - E.g. 'user:*@google.com' would generate violations if there were no users with domain @google.com.
  - E.g. 'group:*@google.com' would generate violations if there were no groups with domain @google.com.
  - E.g. 'user:myuser111@google.com' would generate violations if user:myuser111@google.com does not exist.

### Also included in this PR:

- Sample constraint
- Tests

### CODE REVIEW

**Comment by hshin-g** [comment]
@morgante @briantkennedy Could this get a review from one of you? Thanks!

File: validator/iam_required_bindings_test.rego [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# limitations under the License.
+#
+
+package templates.gcp.GCPIAMRequiredBindingsConstraintV1
+
+import data.test.fixtures.iam_required_bindings.assets as fixture_assets
+import data.test.fixtures.iam_required_bindings.constraints as fixture_constraints
+
+# Find all violations on our test cases
+find_violations[violation] {
```
Comment (briantkennedy) [inline validator/iam_required_bindings_test.rego]: I think this might be able to leverage some of test_utils.rego

Comment (hshin-g) [reply validator/iam_required_bindings_test.rego]: Have changed- PTAL.

Comment (briantkennedy) [reply validator/iam_required_bindings_test.rego]: I think you can also remove find_violations

Comment (hshin-g) [reply validator/iam_required_bindings_test.rego]: Thanks for the catch! Have fixed.

**Pull request merged**